src/analysis/ContextBar.ts

```diff
@@ -123,13 +123,17 @@ export const ContextBar = ({
   const computeRegion = Utils.cond(
     [isGoogleWorkspace(workspace), () => getRegionInfo(googleBucketLocation, googleBucketType).computeRegion],
     [isAzureWorkspace(workspace), () => azureContainerRegion],
-    () => null
-  );
+    () => {
+      throw new Error('Unable to determine cloud provider for workspace');
+    }
+  )!;
```

Comment:
We could consider returning/handling 'UNKNOWN'. There are unfortunately cases where a workspace has failed to delete, but after the Azure cloud context has been deleted. What happens if the error does get thrown here?

Reply:
It looked like downstream code was not prepared to handle the possibility of `computeRegion` or `location` being null. Since `cond` currently doesn't have types, `computeRegion` and `location` here are typed `any`.

Thus, I think this case would cause errors now, but they might not be as loud as throwing an error here.